src/main/java/org/ays/auth/service/impl/AysRoleReadServiceImpl.java

```diff
@@ -31,6 +33,20 @@ class AysRoleReadServiceImpl implements AysRoleReadService {
     private final AysIdentity identity;
 
 
+    /**
+     * Retrieves all roles.
+     * <p>
+     * This method retrieves all roles from the data source and returns them as a set.
+     * </p>
+     *
+     * @return a set of {@link AysRole} objects representing all roles.
+     */
+    @Override
+    public Set<AysRole> findAll() {
+        return new HashSet<>(roleReadPort.findAll());
+    }
```

Comment:
Set dönmek yerine `List` dönebiliriz. MapStruct kütüphanesinde fark ettiğimiz bir problem vardı. Bu sebeple spesifik ihtiyacımız olmadığı sürece `Set` yerine `List` kullanmamız daha sağlıklı olabilir.

Ek olarak yazdığın kodda zaten `roleReadPort.findAll()` metodu `Set` dönüyordu, tekrardan `Set` dönüşümü yapmamamız gerekir.

```suggestion
     * @return a list of {@link AysRole} objects representing all roles.
     */
    @Override
    public List<AysRole> findAll() {
        return roleReadPort.findAll();
    }
```

Reply:
değişikliği yaptım